src/frontend/efiling-frontend/src/domain/authentication/services/AuthService.js

```diff
@@ -56,10 +56,8 @@ export async function getBCSCUserInfo() {
   const bcscToken = await getBCSCUserInfoJWT(brokerToken);
 
   const userInfo = decodeJWT(bcscToken);
-  const firstName = userInfo.given_name ? userInfo.given_name : "";
-  let middleName = userInfo.given_names ? userInfo.given_names : "";
-  middleName = middleName.replace(firstName, "").trim();
+  const givenNames = userInfo.given_names ? userInfo.given_names : "";
   const lastName = userInfo.family_name ? userInfo.family_name : "";
 
-  return { firstName, middleName, lastName };
+  return { givenNames, lastName };
```

Comment:
if we want to go one step further, lastName -> surName to match the keycloak object, since you've done nearly the same with givenNames, but I'm nitpicking at this point.

Reply:
Yeah, good.  I did notice BCSC no longer returns firstName.
Commited patch to return ( givenNames, lastname ).